onnxscript/function_libs/torch_lib/ops/nn.py

```diff
@@ -658,16 +658,141 @@ def aten_huber_loss_backward(
     raise NotImplementedError()
 
 
+def _get_im2col_indices_along_dim(
+    input_d: int,
+    kernel_size_d: int,
+    dilation_d: int,
+    padding_d: int,
+    stride_d: int,
+):
+    # Input is always 4-D (N, C, H, W)
+    # Calculate indices of sliding blocks along spatial dimension
+    # Slide kernel over input each dim d:
+    # each dimension d ranges from 0 to input[d]+2xpadding[d]-dilation[d]x(kernel_size[d]-1)
+    # with steps = stride
+
+    blocks_d = op.Add(input_d, padding_d * 2)
+    blocks_d = op.Sub(blocks_d, (dilation_d * (kernel_size_d - 1)))
```

Comment:
Why can't the above two be combined into one op? Only input_d is dynamic, right?